File: packages/ts-sdk/src/treedx/adapters/federation/mirrors.ts
import { jsonRequest, segment, type TreeDxAdapterContext } from '../runtime/common.js';

export class MirrorsAdapter {
  constructor(private readonly context: TreeDxAdapterContext) {}
  list(repoId: string): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', `/api/v1/repos/${segment(repoId)}/mirrors`); }
  upsert(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/mirrors`, input); }
  sync(repoId: string, mirrorId: string, input?: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/mirrors/${segment(mirrorId)}/sync`, input); }
  health(repoId: string, mirrorId: string, input?: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/mirrors/${segment(mirrorId)}/health`, input); }
  promote(repoId: string, mirrorId: string, input?: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/repos/${segment(repoId)}/mirrors/${segment(mirrorId)}/promote`, input); }
}

File: packages/elixir-sdk/lib/treedx/adapters/federation/mirrors.ex
defmodule TreeDxSdk.Mirrors do
  @moduledoc false
  alias TreeDxSdk.Adapters.Common

  def list(client, repo_id) do
    Common.json_request(
      client,
      :get,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/mirrors",
      nil,
      %{}
    )
  end

  def upsert(client, repo_id, body) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/mirrors",
      body,
      %{}
    )
  end

  def sync(client, repo_id, mirror_id, body \\ %{}) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <>
        Common.segment(repo_id) <> "/mirrors/" <> Common.segment(mirror_id) <> "/sync",
      body,
      %{}
    )
  end

  def health(client, repo_id, mirror_id, body \\ %{}) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <>
        Common.segment(repo_id) <> "/mirrors/" <> Common.segment(mirror_id) <> "/health",
      body,
      %{}
    )
  end

  def promote(client, repo_id, mirror_id, body \\ %{}) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <>
        Common.segment(repo_id) <> "/mirrors/" <> Common.segment(mirror_id) <> "/promote",
      body,
      %{}
    )
  end
end
